### PR TITLE
samples: echo_server: Add a full network stack overflow

### PR DESCRIPTION
**Pull request by ssharks**
Some features like IP fragmenting are default disabled to keep the network stack compact for small devices.
In bigger applications as well as for the Maxwell Pro tests it required to have the full functionality enabled. This can also work as example for new projects.

Signed-off-by: Sjors Hettinga <s.a.hettinga@gmail.com>

### CODE REVIEW

**Comment by ssharks** [comment]
For maintainability we would like to have the config that is for example used for the Maxwell pro tests in the Zephyr repo. Simultaneously this can work as example for projects.
It is an initial attempt, feel free to comment on what is should become.

@hakehuang: I took your config from https://github.com/hakehuang/zephyr/blob/tcp_ip_testing_maxwell/samples/net/sockets/echo_server/overlay-maxwell.conf
Could you work with the default IP adresses that are provided in:
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/sockets/echo_server/prj.conf
CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.0.2.1"
CONFIG_NET_CONFIG_PEER_IPV4_ADDR="192.0.2.2"
This would make the config more generic.

I also notices you increase the stacks of pretty much all threads. Do you know which on of them are really needed.

@nordicjm: Are these reasonable settings for fully applying the fragmenting functionality? Or is it more convenient to make the default settings of the IP fragmenting works sufficiently in these applications?

File: samples/net/sockets/echo_server/overlay-full_network_stack.conf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# enable fragmenting
+CONFIG_NET_IPV4_FRAGMENT=y
+CONFIG_NET_IPV4_FRAGMENT_MAX_COUNT=16
+CONFIG_NET_IPV4_FRAGMENT_TIMEOUT=15
+CONFIG_NET_IPV4_FRAGMENT_MAX_PKT=8
+
+CONFIG_NET_IPV6_FRAGMENT=y
+CONFIG_NET_IPV6_FRAGMENT_MAX_COUNT=16
+CONFIG_NET_IPV6_FRAGMENT_TIMEOUT=15
+CONFIG_NET_IPV6_FRAGMENT_MAX_PKT=8
```
Comment (nordicjm) [inline samples/net/sockets/echo_server/overlay-full_network_stack.conf:22]: I have no clue how this test system works, but what the configuration here is set to is to allow up to 16 packet re-assemblies at a time, for which there are a total of 8 buffers that can be used, shared for all of these (having a higher MAX_COUNT than MAX_PKT doesn't really make sense). So if you have an MTU of 1500 (i.e. ethernet) and these settings, let's assume 1 connection, you could get 8 * 1500 byte fragments to reassemble a 12,000 byte packet, or you could get 8 * 192 byte fragments to reassemble a 1,536 byte packet, but you would not be able to have 9 fragments. Therefore you need to know what the device that is the tester is using, if for example it is using 64 byte packets and trying to test the maximum size packet of 65KB then you would need to reassemble 1024 fragments - which for a small embedded device is really not going to happen. Likewise, is the tester using multiple concurrent connections or just 1?

Comment (hakehuang) [reply samples/net/sockets/echo_server/overlay-full_network_stack.conf:22]: below is my maxwell test configuration with echo server

```
	1. Setup net-tools for bridging
git clone https://github.com/hakehuang/net-tools.git

	./net-maxwell-setup.sh

	2. Build zephyr qemu and run

           cd zephyr
           source ./zephyr-env.sh
           west update
	cd samples/net/socket/echo_server
	west build -b qemu_x86 -- -DCONF_FILE=prj.conf -DOVERLAY_CONFIG="overlay-e1000.conf;overlay-full_network_stack.conf" 
	west build -t run

```

Comment (rlubos) [reply samples/net/sockets/echo_server/overlay-full_network_stack.conf:22]: Looking at the test results, the MTU ranges from 8 to 512 bytes. The question here is, how many fragments can individual test send, the reports aren't clear here, for example:
> IPv6 Fragment Fragment: 8 MTU. Order: N ... 3 2 1. Overlap: None. error

@hakehuang Any chance to get information what's the value of N, i.e. how many fragments can test send? Is this value constant for all tests, or is it determined individually per test case? I think that the only way to have reliable configuration for tests is to know what is the maximum expected number of fragments.

Comment (hakehuang) [reply samples/net/sockets/echo_server/overlay-full_network_stack.conf:22]: @rlubs, from the netcap, looks like it wraps to `z`.

**Comment by carlescufi** [comment]
@rlubos ping

**Review by rlubos — APPROVED** [review]
Works for me, if it turns out we need up update number of fragments supported, we could do it later.

**Review by nordicjm — APPROVED** [review]

**Review by jukkar — APPROVED** [review]
LGTM

**Pull request merged**